src/cryptonote_basic/connection_context.h

```diff
@@ -66,7 +67,15 @@ namespace cryptonote
     uint16_t m_rpc_port;
     uint32_t m_rpc_credits_per_hash;
     bool m_anchor;
-    //size_t m_score;  TODO: add score calculations
+
+    // when we first get a block from a peer, we pick a random other peer,
+    // and we will not relay the block to it, but instead wait to see if we
+    // do get it from that peer. If we do not get this block after a set
+    // time delay, we subtract 1 to its score. If we do, we add 1.
+    // We ban peers with a score less than a threshold
```

Comment:
How does this work with blocks on a fork? Can a peer submit a pre-computed list of blocks and make the node drop all other peers since they don't know of them?

Reply:
Yes.

Reply:
er, no: a peer is selected for checking only if the block was added to the main chain.